modules/core/keeper/keeper_test.go

```diff
@@ -0,0 +1,124 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	"github.com/stretchr/testify/suite"
+
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradekeeper "github.com/cosmos/cosmos-sdk/x/upgrade/keeper"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	ibchost "github.com/cosmos/ibc-go/v3/modules/core/24-host"
+	ibckeeper "github.com/cosmos/ibc-go/v3/modules/core/keeper"
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+	"github.com/cosmos/ibc-go/v3/testing/simapp"
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+
+	app           *simapp.SimApp
+	stakingKeeper clienttypes.StakingKeeper
+	upgradeKeeper clienttypes.UpgradeKeeper
+	scopedKeeper  capabilitykeeper.ScopedKeeper
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	coordinator := ibctesting.NewCoordinator(suite.T(), 1)
+	chainA := coordinator.GetChain(ibctesting.GetChainID(1))
+
+	suite.app = chainA.App.(*simapp.SimApp)
+
+	suite.stakingKeeper = suite.app.StakingKeeper
+	suite.upgradeKeeper = suite.app.UpgradeKeeper
+	suite.scopedKeeper = suite.app.ScopedIBCKeeper
+}
+
+func (suite *KeeperTestSuite) NewIBCKeeper() {
+	ibckeeper.NewKeeper(
+		suite.app.AppCodec(),
+		suite.app.GetKey(ibchost.StoreKey),
+		suite.app.GetSubspace(ibchost.ModuleName),
+		suite.stakingKeeper,
+		suite.upgradeKeeper,
+		suite.scopedKeeper,
+	)
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+// DummyStakingKeeper implements clienttypes.StakingKeeper used in ibckeeper.NewKeeper
+type DummyStakingKeeper struct {
+	dummyField string
+}
+
+func (d DummyStakingKeeper) GetHistoricalInfo(ctx sdk.Context, height int64) (stakingtypes.HistoricalInfo, bool) {
+	return stakingtypes.HistoricalInfo{}, true
+}
+
+func (d DummyStakingKeeper) UnbondingTime(ctx sdk.Context) time.Duration {
+	return 0
+}
```

Comment:
Thanks for adding these! Could we use "Mock" instead of "Dummy"? I find "Dummy" to be non-neutral language

Reply:
yess